cloudpub/ms_azure/service.py

```diff
@@ -539,21 +543,21 @@ def _publish_preview(self, product: Product, product_name: str) -> None:
             List[ProductSubmission],
             self.filter_product_resources(product=product, resource="submission"),
         )[0]
-        if not self._is_submission_in_preview(submission):
-            log.info(
-                "Submitting the product \"%s (%s)\" to \"preview\"." % (product_name, product.id)
+        if self._is_submission_in_preview(submission):
+            log.info("The product \"%s\" is already set to preview", product_name)
+            return
```

Comment:
These are the untested lines so far.

The difference is that the original code was just testing the `if not case`

Now that I've included the `if` case to log and return it shows up that we have a test cover hole since the beginning